nikola/nikola.py

```diff
@@ -1995,22 +1995,19 @@ def generic_renderer(self, lang, output_name, template_name, filters, file_deps=
         filters is the list of filters (usually site.config['FILTERS']) which will be used to post-process the result.
         file_deps (optional) is a list of additional file dependencies (next to template and its dependencies).
         uptodate_deps (optional) is a list of additional entries added to the task's uptodate list.
-        pre_context (optional) is a dict used as a basis for the template context. Its contents might be overwritten.
-        post_context (optional) is a dict merged into the context before passing it to the renderer.
+        context (optional) a dict used as a basis for the template context. The lang parameter will always be added.
         context_deps_remove (optional) is a list of keys to remove from the context after using it as an uptodate dependency. This should name all keys containing non-trivial Python objects; they can be replaced by adding JSON-style dicts in post_deps_dict.
         post_deps_dict (optional) is a dict merged into the copy of context which is used as an uptodate dependency.
         url_type (optional) allows to override the ``URL_TYPE`` configuration
         """
         utils.LocaleBorg().set_locale(lang)
 
-        file_deps = [] if file_deps is None else file_deps
+        file_deps = copy(file_deps) if file_deps else []
         file_deps += self.template_system.template_deps(template_name)
         file_deps = sorted(list(filter(None, file_deps)))
 
-        context = copy(pre_context) if pre_context else {}
+        context = copy(context) if context else {}
```

Comment:
Yes, in case someone calls this function with the same context for different languages in a row. That person will be quite surprised if they all end up with the same `lang` value in their context.